source/public/scripts/main.journal.js

```diff
@@ -0,0 +1,127 @@
+    window.onclick = function (event) {
+      if (event.target == modal) {
+        modal.style.display = "none";
+      }
+    }
+
+    // Save the task when the "Save" button is clicked
+    saveBtn.onclick = function () {
+      // Get task details from modal input fields
+      var taskDesc = document.getElementById("task-desc").value;
+      var taskType = document.getElementById("task-type").value;
+      var taskProject = document.getElementById("task-project").value;
+
+      // Retrieve existing tasks from local storage or initialize as empty array
+      var tasks = JSON.parse(localStorage.getItem("tasks")) || [];
+
+      if (editingIndex >= 0) {
+        // If editing an existing task, update its details
+        tasks[editingIndex] = {
+          name: taskDesc,
+          type_tag: taskType,
+          project_tag: taskProject,
+          completed: tasks[editingIndex].completed // Preserve the completed status
+        };
+      } else {
+        // If adding a new task, create a new task object and add to the tasks array
+        var task = {
+          name: taskDesc,
+          type_tag: taskType,
+          project_tag: taskProject,
+          completed: false // New tasks are not completed by default
+        };
+        tasks.push(task);
+      }
+
+      // Save the updated tasks array to local storage
+      localStorage.setItem("tasks", JSON.stringify(tasks));
+
+      // Refresh the displayed tasks and hide the modal
+      displayTasks();
+      modal.style.display = "none";
+    }
+
+    // Hide the modal when the "Cancel" button is clicked
+    cancelBtn.onclick = function () {
+      modal.style.display = "none";
+    }
+
+    // Delete the current task when the "Delete" button in the modal is clicked
+    deleteBtnModal.onclick = function () {
+      var tasks = JSON.parse(localStorage.getItem("tasks")) || [];
+      if (editingIndex >= 0) {
+        // Remove the task at the editingIndex from the tasks array
+        tasks.splice(editingIndex, 1);
+        // Save the updated tasks array to local storage
+        localStorage.setItem("tasks", JSON.stringify(tasks));
+        // Refresh the displayed tasks and hide the modal
+        displayTasks();
+        modal.style.display = "none";
+      }
+    }
+
+    // Function to display tasks from local storage
+    function displayTasks() {
+      var tasks = JSON.parse(localStorage.getItem("tasks")) || [];
+      taskContainer.innerHTML = ""; // Clear the current task display
+      tasks.forEach(function (task, index) {
+        // Create a new task item element
+        var taskElement = document.createElement("div");
+        taskElement.className = "task-item";
+        taskElement.innerHTML = `
+          <input type="checkbox" id="task-${index}" ${task.completed ? 'checked' : ''}>
+          <label for="task-${index}">Task ${index + 1}: ${task.name}, Type: ${task.type_tag}, Project: ${task.project_tag}</label>
```

Comment:
I don't think it's necessary for us to display the task number. Also it would be helpful to put the type and project tags in separate `div`s since we're planning to display them more like small badges - see the wireframe:
 
![journal_page](https://github.com/cse110-sp24-group33/cse110-sp24-group33/assets/61530252/bbeae804-6be0-4233-84b6-bfdec9792d05)

Reply:
Also then maybe instead of just setting the `innerHTML` it would be better for readability to create new elements and append each into the `div`